### PR TITLE
Move tags into the profile metadata

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

**Comment by robbrockbank** [comment]
Part of the fix for #251

File: lib/api/profile.go [hunk 1]
```diff
@@ -36,6 +36,9 @@ type ProfileMetadata struct {
 	// The name of the endpoint.
 	Name string `json:"name,omitempty" validate:"omitempty,name"`
 
+	// A list of tags that are applied to each endpoint that references this profile.
```
Comment (caseydavenport) [inline lib/api/profile.go]: We should mark this as deprecated in the code, yeah?

File: lib/client/profile_e2e_test.go [hunk 2]
```diff
@@ -48,14 +48,14 @@ import (
 var profileSpec1 = api.ProfileSpec{
 	IngressRules: []api.Rule{testutils.InRule1, testutils.InRule2},
 	EgressRules:  []api.Rule{testutils.EgressRule1, testutils.EgressRule2},
-	Tags:         []string{"profile1-tag1", "profile1-tag2"},
 }
+var tags1 = []string{"profile1-tag1", "profile1-tag2"}
```
Comment (caseydavenport) [inline lib/client/profile_e2e_test.go]: Any reason we can't inline these in the tests below rather than creating these special variables?

**Comment by caseydavenport** [comment]
Couple of minor comments, then LGTM.

**Comment by caseydavenport** [comment]
I'm assuming that old profiles will currently just be silently allowed by calicoctl without any warning that tags have moved, but that this will be fixed by another PR in calicoctl to add validation for malformed / unsupported fields?

**Comment by robbrockbank** [comment]
@caseydavenport re your last point.  Yes old profile YAML will silently ignore the old tag field until I add in the updated validation.

**Pull request merged**